app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -102,12 +102,12 @@ def realm_field(f, can_override = false, override = false)
 
   def interesting_klasses(obj)
     classes    = obj.all_puppetclasses
-    classes_ids = classes.reorder('').pluck('puppetclasses.id')
+    classes_ids = (classes.reorder('').pluck('puppetclasses.id') + obj.puppetclasses.map(&:id)).uniq.sort
```

Comment:
@ohadlevy: for the same reason we call `reorder`? I am convinced we do not need it but there must be some reason it is there, perhaps I just do not see it...